cpp/include/cudf/io/types.hpp

```diff
@@ -157,6 +160,7 @@ struct sink_info {
   io_type type = io_type::VOID;
   std::string filepath;
   std::vector<char>* buffer = nullptr;
+  std::shared_ptr<cudf::io::data_sink> user_sink = nullptr;
```

Comment:
Are the `filepath/buffer` members still necessary if the `data_sink` base class can replace the need for those members?

Said another way, isn't it sufficient to just have a `data_sink*` in `sink_info`?

Reply:
Almost.  This struct currently gets passed down one layer and then these params get forwarded to a function that creates the data_sink.  I think the python guys may poke these values directly though.